web-client/src/client/lib.rs

```diff
@@ -552,7 +552,21 @@ impl Client {
             .insert(hash.clone(), sender);
 
         // Actually send the transaction
-        consensus.send_transaction(tx.native()).await?;
+        consensus
+            .send_transaction(tx.native())
+            .await
+            .map_err(|error| {
+                // Unsubscribe from any address we subscribed to, without caring about the result
+                if let Some(address) = subscribed_address.clone() {
+                    let owned_consensus = consensus.clone();
+                    spawn_local(async move {
+                        let _ = owned_consensus
+                            .unsubscribe_from_addresses(vec![address], 1)
+                            .await;
+                    });
+                }
+                error
+            })?;
```

Comment:
Mmm, my $0.02 is that this is very error prone if we add more code between these lines and the place where we do the unsusbscribe below plus we are duplicating code. Maybe we could store this in a `struct` and implement `Drop` for this `struct`?

Reply:
Oh, that's brilliant! 👍

Reply:
You can also see the pattern in 297a99a0b1531b703b6332132842410d51cf33f9.

Reply:
Done.